uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/models/instances/sickness-leave-request-instance_test.js

```diff
@@ -0,0 +1,163 @@
+define([
+  'mocks/helpers/helper',
+  'mocks/apis/leave-request-api-mock',
+  'mocks/apis/option-group-api-mock',
+  'leave-absences/shared/models/instances/sickness-leave-request-instance',
+], function (helper) {
+  'use strict';
+
+  describe('SicknessRequestInstance', function () {
+    var expectedError, instance, LeaveRequestAPI, $provide, promise, $q, requestData, $rootScope;
+
+    beforeEach(module('leave-absences.models.instances', 'leave-absences.mocks', function (_$provide_) {
+      $provide = _$provide_;
+    }));
+
+    beforeEach(inject(function (_LeaveRequestAPIMock_, _OptionGroupAPIMock_) {
+      $provide.value('LeaveRequestAPI', _LeaveRequestAPIMock_);
+      $provide.value('OptionGroup', _OptionGroupAPIMock_);
+    }));
+
+    beforeEach(inject([
+      'SicknessRequestInstance', '$rootScope', 'LeaveRequestAPI', '$q',
+      function (_SicknessRequestInstance_, _$rootScope_, _LeaveRequestAPI_, _$q_) {
+        instance = _SicknessRequestInstance_.init({}, false);
+        $rootScope = _$rootScope_;
+        LeaveRequestAPI = _LeaveRequestAPI_;
+        $q = _$q_;
+
+        spyOn(LeaveRequestAPI, 'create').and.callThrough();
+        spyOn(LeaveRequestAPI, 'update').and.callThrough();
+        spyOn(LeaveRequestAPI, 'isValid').and.callThrough();
+      }
+    ]));
+
+    describe('init', function () {
+      it('sickness request', function () {
+        expect(instance).toBeDefined();
+      });
+    });
+
+    describe('create()', function () {
+      beforeEach(function () {
+        requestData = helper.createRandomSicknessRequest();
+        instance = instance.init(requestData, false);
+        promise = instance.create();
+      });
+
+      afterEach(function () {
+        //to excute the promise force an digest
+        $rootScope.$apply();
+      });
+
+      it('calls equivalent API method', function () {
+        promise.then(function () {
+          expect(LeaveRequestAPI.create).toHaveBeenCalled();
+        });
+      });
+
+      it('id is appended to instance', function () {
+        expect(instance.id).not.toBeDefined();
+        promise.then(function () {
+          expect(instance.id).toBeDefined();
+          expect(instance.id).toEqual(jasmine.any(String));
+        });
+      });
+
+      describe('when one mandatory field is missing', function () {
```

Comment:
This test should not be part of this file, as SicknessInstance does not deal with error handling. Any code which is not part of a file(abstracted) should also not be tested as part of that file. Otherwise for a single code we need test multiple times.